_sips/minutes/2020-03-11-minutes.md

```diff
@@ -0,0 +1,220 @@
+participate because of coronavirus-related travel restrictions and disruptions.
+
+## Proceedings
+
+### Expectations
+
+* Martin: The plan is still to release Scala 3 this year, at the end of December.
+* Martin: However, TASTy won't be stable until 3.1 - it won't be a 3.0 promise.
+* Martin: Ideally the features would be voted on with the following 3 outcomes
+
+* accepted
+* accepted, but put behind some kind of "preview" flag, like Java is doing
+* rejected
+* postponed to after 3.0
+
+### Review a "easy" dotty features/changes
+
+This is the list, with any notable comments added.
+
+* Context functions
+* Drop auto application
+* Drop auto tupling
+    * Martin: this is probably the most breaking change, so weakly against it
+    * Seth: could it be behind `-Scala2`?  Martin: no, that would be difficult
+    * There's some difficulties in implementing a rewrite rule for it
+* Drop class shadowing
+    * Seb: this is binary breaking for existing libraries
+    * Seth: let's make it strong (not under -Xlint) w/ opt-out
+* Drop compound types
+* Drop DelayedInit
+    * most of the committee is for it
+    * but a number of usages of it have been reported
+    * Guillaume: the implementation isn't a big deal, but it affects TASTy
+    * ScalaTest, specs2, and cats-effect use it
+    * Let's drop it and hopefully someone can write a compiler plugin for it
+    * Lukas: DelayedInit was the source of lots of bugs in nsc, most fixed by now, though
+* Drop do..while
+    * added because of indentation syntax
+    * how many more like this one?
+* Drop early initializers
+* Drop existential types
+    * Guillaume: can be replaced by using opaque types, with explicit wrap and unwrapping
+* Drop non-local returns
+* Drop package objects
+    * Lukas: what's wrong with package objects, or it having parents?  Don't exports suffer the same problems?
+    * Seb: package objects are the only thing where you can extend something "in you".  Exports don't have that.
+    * Lukas: what about implicit prioritisation w/ exports?
+    * Martin: there's a change in implicit resolution so you can add dummy implciit parameters and prioritise those (with companion object prioritisation)
+* Drop procedure syntax
+    * Guillaume: it's dead
+* Drop Symbol literals
+    * Seb: it's basically done
+* Drop XML literals
+    * Guillaume: 3.0 supports XML literals (under `-Scala2` flag?)
+    * Seb: it doesn't leak into binaries/TASTy, can remove later
+    * Nic: we have a replacement but just without unapply/pattern matching
+    * Martin: and that isn't used much, so you'd just need to do something else there
+* Eta expansion
+    * no `_` any more
+    * Gui: issue with println, you `println(myMethod)` and it's not a type error (b/c no args) it prints the function
+    * Martin: multi-verse equality fixes `==` for functions, previously always `false`
+* Implicit conversions with Conversion
+    * Gui: from threads, used a lot by DSL libraries
```

Comment:
add:
* Seth: but that was in reference to the explicit-only option, which doesn't have committee support. most DSLs should be fine under the proposal